### PR TITLE
Disable the 8259 PICs during kernel startup.

### PR DESCRIPTION
**Pull request by andrisaar**

### CODE REVIEW

File: oak_restricted_kernel/src/interrupts.rs [hunk 1]
```diff
@@ -350,3 +351,65 @@ pub unsafe fn init_idt(double_fault_stack_index: u16) {
...
+
+    let mut pic0 = Pic::new(&io_port_factory, 0x20);
+    let mut pic1 = Pic::new(&io_port_factory, 0xA0);
+
+    // The initialization process is documented in https://wiki.osdev.org/8259_PIC
+    // Tell the PICs we're going to start intializing them.
+    pic0.write_command(0x11)?; // ICW1_INIT | ICW1_ICW4
+    pic1.write_command(0x11)?;
+
+    // Byte 1: interrupt offsets
```
Comment (conradgrobler) [inline oak_restricted_kernel/src/interrupts.rs:398]: nit: comments should end with a `.`

File: oak_restricted_kernel/src/interrupts.rs [hunk 2]
```diff
@@ -350,3 +351,65 @@ pub unsafe fn init_idt(double_fault_stack_index: u16) {
...
+
+    // Byte 2: chanining between PICs
+    pic0.write_data(4)?; // master: there's a slave PIC at IRQ 2
+    pic1.write_data(2)?; // slave: your cascade identity
+
+    // Byte 3: operation mode
+    pic0.write_data(0x01)?; // ICW4_8086
+    pic1.write_data(0x01)?;
+
+    // and finally, mask all interrupts as we're not going to use the PICs.
```
Comment (conradgrobler) [inline oak_restricted_kernel/src/interrupts.rs:410]: nit: this comment should start with a capital.

File: oak_restricted_kernel/src/interrupts.rs [hunk 3]
```diff
@@ -350,3 +351,65 @@ pub unsafe fn init_idt(double_fault_stack_index: u16) {
...
+    // Tell the PICs we're going to start intializing them.
+    pic0.write_command(0x11)?; // ICW1_INIT | ICW1_ICW4
+    pic1.write_command(0x11)?;
+
+    // Byte 1: interrupt offsets
+    pic0.write_data(0x20)?; // PIC0 interrupts start at vector 32
+    pic1.write_data(0x28)?; // PIC1 interrupts start at vector 40
+
+    // Byte 2: chanining between PICs
+    pic0.write_data(4)?; // master: there's a slave PIC at IRQ 2
```
Comment (conradgrobler) [inline oak_restricted_kernel/src/interrupts.rs:403]: perhaps try to avoid terms such as master and slave, and use something like leader and follower.

**Review by conradgrobler — APPROVED** [review]

File: oak_restricted_kernel/src/interrupts.rs (hunk 3 above)
Comment (andrisaar) [reply oak_restricted_kernel/src/interrupts.rs]: Legacy technology, legacy terminology :)

But I've reworded it to not have any references like that.

**Pull request merged**